ludwig/explain/base.py

```diff
@@ -0,0 +1,74 @@
+from abc import ABCMeta, abstractmethod
+from typing import List, Tuple
+
+import pandas as pd
+
+from ludwig.api import LudwigModel
+from ludwig.constants import BINARY, CATEGORY, TYPE
+from ludwig.explain.util import Explanation, prepare_data
+from ludwig.utils.torch_utils import get_torch_device
+
+DEVICE = get_torch_device()
+
+
+class Explainer(metaclass=ABCMeta):
```

Comment:
We could call this file `explainer.py` since its main export is the `Explainer` class